packages/wrangler/src/pipelines/cli/create.ts

```diff
@@ -14,299 +14,284 @@ import {
 	BindingSource,
 	HttpSource,
 	PipelineUserConfig,
 	Source,
 } from "../client";
-import type { Argv } from "yargs";
 
-export function addCreateOptions(yargs: Argv<CommonYargsOptions>) {
-	return (
-		yargs
-			.positional("pipeline", {
-				describe: "The name of the new pipeline",
-				type: "string",
-				demandOption: true,
-			})
-			// Sources
-			.group(
-				["source", "require-http-auth", "cors-origins"],
-				`${chalk.bold("Source settings")}`
-			)
-			.option("source", {
-				type: "array",
-				describe:
-					"Space separated list of allowed sources. Options are 'http' or 'worker'",
-				default: ["http", "worker"],
-				demandOption: false,
-			})
-			.option("require-http-auth", {
-				type: "boolean",
-				describe:
-					"Require Cloudflare API Token for HTTPS endpoint authentication",
-				default: false,
-				demandOption: false,
-			})
-			.option("cors-origins", {
-				type: "array",
-				describe:
-					"CORS origin allowlist for HTTP endpoint (use * for any origin). Defaults to an empty array",
-				demandOption: false,
-				coerce: validateCorsOrigins,
-			})
+export const pipelinesCreateCommand = createCommand({
+	metadata: {
+		description: "Create a new pipeline",
+		owner: "Product: Pipelines",
+		status: "open-beta",
+	},
+	args: {
+		pipeline: {
+			describe: "The name of the new pipeline",
+			type: "string",
+			demandOption: true,
+		},
 
-			// Batching
-			.group(
-				["batch-max-mb", "batch-max-rows", "batch-max-seconds"],
-				`${chalk.bold("Batch hints")}`
-			)
-			.option("batch-max-mb", {
-				type: "number",
-				describe:
-					"Maximum batch size in megabytes before flushing. Defaults to 100 MB if unset. Minimum: 1, Maximum: 100",
-				demandOption: false,
-				coerce: validateInRange("batch-max-mb", 1, 100),
-			})
-			.option("batch-max-rows", {
-				type: "number",
-				describe:
-					"Maximum number of rows per batch before flushing. Defaults to 10,000,000 if unset. Minimum: 100, Maximum: 10,000,000",
-				demandOption: false,
-				coerce: validateInRange("batch-max-rows", 100, 10_000_000),
-			})
-			.option("batch-max-seconds", {
-				type: "number",
-				describe:
-					"Maximum age of batch in seconds before flushing. Defaults to 300 if unset. Minimum: 1, Maximum: 300",
+		source: {
+			type: "array",
+			describe:
+				"Space separated list of allowed sources. Options are 'http' or 'worker'",
+			default: ["http", "worker"],
+			demandOption: false,
+			group: `${chalk.bold("Source settings")}`,
+		},
+		"require-http-auth": {
+			type: "boolean",
+			describe:
+				"Require Cloudflare API Token for HTTPS endpoint authentication",
+			default: false,
+			demandOption: false,
+			group: `${chalk.bold("Source settings")}`,
+		},
+		"cors-origins": {
+			type: "array",
+			describe:
+				"CORS origin allowlist for HTTP endpoint (use * for any origin). Defaults to an empty array",
+			demandOption: false,
+			coerce: validateCorsOrigins,
+			group: `${chalk.bold("Source settings")}`,
```

Comment:
NIT: shouldn't we do the formatting in the createCommand internals for groups rather than hard coding it all over the place in each command?

Reply:
caca69f01